src/Framework/TableAccessor.php

```diff
@@ -0,0 +1,301 @@
+<?php
+namespace Give\Framework;
+
+use http\Exception\InvalidArgumentException;
+use wpdb;
+
+/**
+ * Class Table
+ * @package Give\Database\Tables
+ *
+ * @since 2.9.0
+ */
+abstract class TableAccessor {
+	/**
+	 * @since 2.9.0
+	 * @var Table
+	 */
+	protected $table;
+
+	/**
+	 * @since 2.9.0
+	 * @var wpdb
+	 */
+	private $db;
+
+	/**
+	 * Table constructor.
+	 *
+	 * @param  Table  $table
+	 */
+	public function __construct( Table $table ) {
+		$this->table = $table;
+		$this->db    = $this->table->getDb();
+	}
+
+	/**
+	 * Retrieve a row by the primary key
+	 *
+	 * @since  2.9.0
+	 * @access public
+	 *
+	 * @param  int $primaryKeyValue Primary key value.
+	 *
+	 * @return object
+	 */
+	public function get( $primaryKeyValue ) {
+		return $this->db->get_row(
+			$this->db->prepare(
+				"
+			SELECT * FROM {$this->table->getName()}
+			WHERE {$this->table->getPrimaryKey()} = %s LIMIT 1;",
+				$primaryKeyValue
+			)
```

Comment:
Slight formatting tweak.
```suggestion
			$this->db->prepare("
					SELECT * FROM {$this->table->getName()}
					WHERE {$this->table->getPrimaryKey()} = %s LIMIT 1;
				",
				$primaryKeyValue
			)
```